libraries/provider_systemd_unit.rb

```diff
@@ -0,0 +1,50 @@
+require 'chef/provider/lwrp_base'
+require_relative 'helpers'
+
+class Chef::Provider
+  class SystemdUnit < Chef::Provider::LWRPBase
+    use_inline_resources
+
+    def whyrun_supported?
+      true
+    end
+
+    provides :systemd_unit
+    Systemd::Helpers.unit_types.each do |unit_type|
+      provides "systemd_#{unit_type}".to_sym
+    end
+
+    action :create do
+      unit_path = Systemd::Helpers.unit_path(new_resource)
+
+      execute 'reload-sytemd' do
+        command 'systemctl daemon-reload'
+        action :nothing
+        subscribes :run, "file[#{unit_path}]", :immediately
+      end
+
+      converge_by "Creating systemd unit: #{unit_path}" do
```

Comment:
aha, on further reading, it seems i misunderstood the documentation. i initially thought that was how to add whyrun support (wrap the resource that acts on the system in a converge_by), but i see now that that's not required for lightweight providers since the whyrun support's provided by the platform resource (in this case the file resource).